### PR TITLE
feat: Display current announcements

### PR DESCRIPTION
**Pull request by jcbcapps**
# SC-1083

## Proposed changes
Adds filtering to only display announcements up until the current date, so that anything with a published date in the future is not shown.

---

## Reviewer notes
You will need to also have `sc-1083/announcement-scheduler` checked out in the CMS to view these changes

Also please review:
- PR for updating CMS: https://github.com/USSF-ORBIT/ussf-portal-cms/pull/262
- PR for e2e test: https://github.com/USSF-ORBIT/ussf-portal/pull/194

## Setup
Run everything as usual
In the CMS: `yarn services:up && yarn dev`
In the portal: `yarn dev`

Create an announcement with a published date in the future, and make sure that it does not display in the portal.

---

### As the original developer, I have

- [ ] Met the acceptance criteria, or will meet them in subsequent PRs or stories
  - ... 
- [ ] Created new stories in Storybook if applicable
  - [ ] Checked that all Storybook accessibility checks are passing
- [ ] Created/modified automated unit tests in Jest
  - [ ] Including jest-axe checks when UI changes
- [ ] Created/modified automated E2E tests in Cypress
  - [ ] Including cypress-axe checks when UI changes
  - [ ] Checked that the E2E test build is not failing
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)
- [ ] Requested a design review for user-facing changes
- For any new migrations/schema changes:
  - [ ] Followed guidelines for zero-downtime deploys

### As code reviewer(s), I have

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
  - [ ] Checked that the E2E test build is not failing
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] Considered marking this as accepted even if there are small changes needed
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a designer reviewer, I have

- [ ] Checked in the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow
- Performed [a11y testing](https://github.com/trussworks/accessibility/blob/master/sample_a11y_testing_process.md):
  - [ ] Checked responsiveness in mobile, tablet, and desktop
  - [ ] Checked keyboard navigability
  - [ ] Tested with [VoiceOver](https://dequeuniversity.com/screenreaders/voiceover-keyboard-shortcuts) in Safari
  - [ ] Checked VO's [rotor menu](https://github.com/trussworks/accessibility/blob/master/README.md#how-to-use-the-rotor-menu) for landmarks, page heading structure and links
  - [ ] Used a browser a11y tool to check for issues (WAVE, axe, ANDI or Accessibility addon tab for Storybook)

### As a test user, I have

- Run through the [Test Script](hhttps://docs.google.com/spreadsheets/d/1eV8UK0aJZ0qrzjnXf5SJ_uRiV7bpsj3yrhRFEkjOvV4/edit?usp=sharing):
  - [x] On commercial internet in IE11 / Edge
  - [x] On commercial internet in Firefox
  - [x] On commercial internet in Chrome
  - [x] On commercial internet in Safari
  - [ ] On NIPR in IE11
  - [ ] On NIPR in Firefox
  - [ ] On NIPR in Chrome
  - [ ] On NIPR in Safari
  - [ ] On a mobile device in Firefox
  - [ ] On a mobile device in Chrome
  - [ ] On a mobile device in Safari
- [ ] Added any anomalous behavior to this PR

---

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -44,14 +43,21 @@ export async function getServerSideProps() {
...
 
   const bookmarks = cmsBookmarks?.bookmarks as BookmarkRecords
 
+  // Filter announcements array to only contain announcements with
+  // a publish date up until now.
+  announcements = announcements.filter(
+    (a: AnnouncementRecord) =>
+      DateTime.fromISO(a.publishedDate) < DateTime.now()
+  )
+
```
Comment (jcbcapps) [inline src/pages/index.tsx:60]: So this works, but I wanted to find a way to do this by updating the GraphQL schema. Looks like I would need to update `extendGraphqlSchema` in the `src/lib/schema.ts` file in the CMS repo? Not sure. Any suggestions?

Comment (gidjin) [reply src/pages/index.tsx:60]: You won't need to do that. You should just need to update the GraphQL query on the client to limit by date. See my changes in `src/operations/cms/queries/getInternalNewsCarouselArticles.ts`. You I believe will need to modify `src/operations/cms/queries/getAnnouncements.ts` to limit by published date as well. Also you will need to pass the current date in as a parameter. Let me know if you have questions.

File: src/pages/news-announcements.tsx [hunk 2]
```diff
@@ -137,8 +138,18 @@ export async function getServerSideProps() {
     data: { articles },
   } = await client.query({
     query: GET_INTERNAL_NEWS_CAROUSEL_ARTICLES,
+    variables: {
+      publishedDate: DateTime.now(),
+    },
```
Comment (gidjin) [inline src/pages/news-announcements.tsx:143]: Here is where I am passing current time to the GraphQL query to limit articles found in the carousel.

**Review by gidjin — CHANGES_REQUESTED** [review]
I agree with you that we should update the query, so marking this as requested changes for now.

**Comment by jbecker01** [comment]
Looks good!

**Review by gidjin — APPROVED** [review]
LGTM!

**Pull request merged**